app/src/main/java/zapsolutions/zap/customView/OnChainFeeView.java

```diff
@@ -0,0 +1,198 @@
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+    private void init() {
+        View view = inflate(getContext(), R.layout.view_onchain_fee, this);
+
+        mTvSendFeeAmount = view.findViewById(R.id.sendFeeOnChainAmount);
+        mTvSendFeeSpeed = view.findViewById(R.id.sendFeeSpeed);
+        mTabLayoutSendFeeSpeed = view.findViewById(R.id.feeSpeedTabLayout);
+        mTvSendFeeDuration = view.findViewById(R.id.feeDurationText);
+        mClSendFeeAmountLayout = view.findViewById(R.id.sendFeeOnChainAmountLayout);
+        mFeeArrowUnitImage = view.findViewById(R.id.feeArrowUnitImage);
+
+        mClSendFeeDurationLayout = view.findViewById(R.id.feeDurationLayout);
+
+        // Toggle tier settings view on amount click
+        mClSendFeeAmountLayout.setOnClickListener(new OnSingleClickListener() {
+            @Override
+            public void onSingleClick(View v) {
+                boolean isFeeDurationVisible = mClSendFeeDurationLayout.getVisibility() == View.VISIBLE;
+                toggleFeeTierView(isFeeDurationVisible);
+            }
+        });
+
+        // Disable fee tier change until successful calculation
+        mClSendFeeAmountLayout.setClickable(false);
+
+        // Set initial block target time
+        setBlockTargetTime(mOnChainFeeTier.getConfirmationBlockTarget());
+
+        // Listen for tier change by user
+        mTabLayoutSendFeeSpeed.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tab) {
+              if(tab.getText() != null) {
+                  if (tab.getText().equals(getResources().getString(OnChainFeeTier.SLOW.getTitle()))) {
+                      setFeeTier(OnChainFeeTier.SLOW);
+                  } else if (tab.getText().equals(getResources().getString(OnChainFeeTier.MEDIUM.getTitle()))) {
+                      setFeeTier(OnChainFeeTier.MEDIUM);
+                  } else if (tab.getText().equals(getResources().getString(OnChainFeeTier.FAST.getTitle()))) {
+                      setFeeTier(OnChainFeeTier.FAST);
+                  }
+              }
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
+
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tab) {
+
+            }
+        });
+    }
+
+    public OnChainFeeTier getFeeTier() {
+        return mOnChainFeeTier;
+    }
+
+    public void setFeeTierChangedListener(FeeTierChangedListener feeTierChangedListener) {
+        mFeeTierChangedListener = feeTierChangedListener;
+    }
+
+    public void onFeeSuccess(String amount) {
+        mTvSendFeeAmount.setText(amount);
+        mClSendFeeAmountLayout.setClickable(true);
+    }
+
+    public void onFeeFailure() {
+        mTvSendFeeAmount.setText(R.string.fee_not_available);
+        mClSendFeeAmountLayout.setClickable(false);
```

Comment:
I'm not sure about this line.
If the fee calculation returns an error, it disables the function to show the fee tier selection. User has to change the amount to trigger new fee calculation.

**My assumption** 
If the calculation returns an error for either fee tier, the amount is invalid and a change in fee tier would not help to get to a valid fee.

**Possibility**
If one is able to change the fee tier (while keeping the same amount), the fee calculation is successful and a payment can be sent.

**What I have observed** 
Using lower transaction amounts (lower than $5) and the fast fee tier, most of the time results in an invalid fee.
After switching to medium/slow fee tier, the fee calculation returns a valid amount (below 1$), but the payment cannot be sent anyways because of transaction output is dust.

So I'm not sure where the error happens here. 
Should we allow the change of fee tier even though there was an invalid fee calculation.

Reply:
I would say yes, allow it. If the payment will not work, there will be an error message after hitting "send".
It is not ideal, but that way one can at least try to change the fee to make it a valid payment.